pages/api/recipesMeals/[nameMeal].js

```diff
@@ -0,0 +1,43 @@
+import axios from 'axios';
+
+const ingredientsWithMeasure = (ingredients, measure) => {
+  const list = ingredients.reduce((accumulator, currentValue, index) => {
+    const ingredientWithMeasure = currentValue + " " + measure[index];
+    return [...accumulator, ingredientWithMeasure]
+  }, []);
+  return list;
+}
+const itens = (recipeDetails, element) => {
+  const list = Array(19).fill(0).reduce((acc, _currentValue, index) => {
```

Comment:
Why is this hardcoded to 19?

Reply:
This api always returns 19 keys of ingredient objects even though not all have values, these keys go from 1 to 20, therefore 19 keys, but they are object keys and not array elements, so I could not go through it, so my idea was create an array of 19 positions to traverse, and with the index of each element of the array access as keys of the objects, since the keys have a pattern of "strIngredient" + number, knowing that it starts with 1, I used index + 1 to traverse .